add open liberty package

cpanato opened the pull request:
xref: https://github.com/chainguard-dev/image-requests/issues/4078

cpanato committed:
add open liberty package

jamonation commented:
@EyeCantCU did it look like this would be buildable with OpenJ9?

EyeCantCU committed:
chore(open-liberty): Small cleanup, expand tests

We don't need the full OpenJDK during tests or to copy the zip out to
the destination directory; just extract it

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(open-liberty): Build for all supported OpenJDK versions

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
Revert "chore(open-liberty): Build for all supported OpenJDK versions"

Requires OpenJDK 17+ at buildtime

This reverts commit 2b134a41cdfc0b565807af68a247daace1460dc4.

EyeCantCU commented:
Upstream builds with 17 and distributes images with all LTS Java releases at runtime - we'll handle this at the image level due to this

EyeCantCU committed:
chore(open-liberty): Try a resource bump

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(open-liberty): Take a stack trace

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(open-liberty): Add Docker scripts and configuration

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(open-liberty): Try refreshing deps

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(open-liberty): Bump commit for 24.0.0.11

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

jamonation commented:
What happens with adding `-x test` to the gradle commands? Not that we want to do that, but does it get things building? If it does, we could narrow down the failing tests and get them sorted out or excluded if they don't run properly in CI.

jamonation committed:
Update open-liberty.yaml

Signed-off-by: Jamon Camisso <jamonation@users.noreply.github.com>

jamonation commented on open-liberty.yaml:
```diff
@@ -0,0 +1,91 @@
+        - openjdk-21-default-jvm
+  pipeline:
+    - runs: |
+        export PATH="/usr/share/java/openliberty/wlp/bin:$PATH"
```
```suggestion
        export PATH="/usr/share/java/open-liberty/wlp/bin:$PATH"
```

jamonation committed:
Update open-liberty.yaml

Signed-off-by: Jamon Camisso <jamonation@users.noreply.github.com>

jamonation commented on open-liberty.yaml:
```diff
@@ -0,0 +1,91 @@
+
+  - working-directory: dev
+    runs: |
+      ./gradlew --refresh-dependencies --stacktrace cnf:initialize -x test
+      ./gradlew --refresh-dependencies --stacktrace assemble -x test
+      ./gradlew --refresh-dependencies --stacktrace releaseNeeded -x test
```
```suggestion
      ./gradlew --refresh-dependencies --stacktrace cnf:initialize
      ./gradlew --refresh-dependencies --stacktrace assemble
      ./gradlew --refresh-dependencies --stacktrace releaseNeeded
```

jamonation replied:
probably don't need javadocs here, but leaving as is

jamonation committed:
Update open-liberty.yaml

Signed-off-by: Jamon Camisso <jamonation@users.noreply.github.com>

jamonation commented on open-liberty.yaml:
```diff
@@ -0,0 +1,91 @@
+package:
```
```suggestion
#nolint:valid-pipeline-git-checkout-tag
package:
```

jamonation committed:
add nolint for open-liberty docker subrepo clone

Signed-off-by: Jamon Camisso <jamonation@users.noreply.github.com>

EyeCantCU committed:
fix(open-liberty): Prevent maven repos from clashing for each arch

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(open-liberty): Use the superior bsdtar

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(open-liberty): Enable parallelism and drop our gradle

The wrapper fetches the needed version of gradle regardless of us
providing it

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

cpanato commented:
🎉
